### PR TITLE
fix: use stop place ID instead of reverse geocoder to show stop place details.

### PR DESCRIPTION
**Pull request by reidzeibel**
closes https://github.com/AtB-AS/kundevendt/issues/18275

~~This will add a fallback to use the stop place ID when the reverse geocoder fails, it should fix the missing departures on Brekstad & Tromsø.~~

Removes the use of `reverse geocoder` to look for stop place details, and instead use the stop place parent ID directly to fetch the details. In this PR, I introduced the use of a new BFF endpoint to look for the stop place parent ID (https://github.com/AtB-AS/atb-bff/pull/340), and use it as the params for the stop place details API.

Also I refactored the `useStopDetailsData` into react query and made the parent ID lookup function with react query.

#### Brekstad

image

image

image

#### Tromsø

image

image

### CODE REVIEW

**Comment by rosvik** [comment]
Like we talked about in person, we can do a journey planner lookup for parent ID instead: [example](https://api.entur.io/graphql-explorer/journey-planner-v3?query=%7B%0A%20%20stopPlace%28id%3A%20%22NSR%3AStopPlace%3A74008%22%29%20%7B%0A%20%20%20%20id%0A%20%20%20%20parent%20%7B%0A%20%20%20%20%20%20id%0A%20%20%20%20%7D%0A%20%20%7D%0A%7D%0A)

**Comment by reidzeibel** [comment]
@rosvik ready for another round of review

File: src/components/map/components/DeparturesDialogSheet.tsx [hunk 1]
```diff
@@ -46,44 +41,43 @@ export const DeparturesDialogSheet = ({
...
-    error: geocoderError,
-    forceRefresh: forceRefreshReverseGeocode,
-  } = useReverseGeocoder({longitude, latitude} || null, ['venue']);
   const appStateStatus = useAppStateStatus();
 
-  const filteredLocations = locations?.filter((location) =>
-    stopPlaceFeature.properties?.name.includes(location.name),
-  );
+  const stopPlaceId =
+    stopPlaceFeature.properties && stopPlaceFeature.properties['id'];
```
Comment (rosvik) [inline src/components/map/components/DeparturesDialogSheet.tsx:47]:
```suggestion
  const stopPlaceId = stopPlaceFeature.properties?.['id'];
```

**Review by rosvik — APPROVED** [review]
Looks great! Just a couple nits. Nice to get the reducer replaced with useQuery ✨ 

Also, since this touches the PlaceScreen, it's nice to add test input in the issue, that loading of stop details in departures is refactored. (Which is used in the map, departures tab, and when adding favorites on the front page)

**Comment by reidzeibel** [comment]
@rosvik added the test acceptance criteria

**Pull request merged**